src/subscribeToQuery/index.ts

```diff
@@ -175,12 +195,18 @@ export async function subscribeToQuery<
     const registration = await req.json();
 
     channelUrl = registration.url;
+    if (onEvent) {
+      onEvent({status: 'connecting', channelUrl, message: 'Received channel URL'});
+    }
   } catch (e) {
-    if (e instanceof Response400Error || e instanceof InvalidResponseError) {
-      throw e;
```

Comment:
If the response is a 4xx, it means that the requested GraphQL query has some errors, and reconnecting won't help solve the situation. The same kinda applies to InvalidResponseError.. if the server is returning something that's not JSON it probably isn't something that can be solved by retrying. For the latter, maybe we can discuss about it :)

Reply:
I've amended the change to only throw on 404s.

Reply:
I closed this PR in favour of #2. Let's continue the review there